src/OpenTelemetry/Internal/SelfDiagnosticsEventListener.cs

```diff
@@ -157,6 +174,22 @@ protected override void OnEventSourceCreated(EventSource eventSource)
         {
             if (eventSource.Name.StartsWith(EventSourceNamePrefix, StringComparison.Ordinal))
             {
+                // If there are EventSource classes already initialized as of now, this method would be called from
+                // the base class constructor before the first line of code in SelfDiagnosticsEventListener constructor.
+                // In this case logLevel is always its default value, "LogAlways".
+                // Thus we should save the event source and enable them later, when code runs in constructor.
+                if (this.eventSourcesBeforeConstructor != null)
+                {
+                    lock (this.eventSourcesBeforeConstructor)
```

Comment:
a dedicated lock obj instead?

Reply:
Make sense. I've updated it. Although I didn't make it a static field like the `EventCounterListener` did, because the lock is [used for an instance variable (`eventSourcesBeforeConstructor`)](https://stackoverflow.com/questions/6891711/difference-between-lockthis-and-a-lock-on-static-object). And using static might [cause an unneeded performance hit](https://stackoverflow.com/questions/6891711/difference-between-lockthis-and-a-lock-on-static-object).